src/server/engine_shard_set.cc

```diff
@@ -602,7 +605,9 @@ void EngineShard::Heartbeat() {
     ttl_delete_target = kTtlDeleteLimit * double(deleted) / (double(traversed) + 10);
   }
 
-  ssize_t redline = (max_memory_limit * kRedLimitFactor) / shard_set->size();
+  ssize_t eviction_redline = (max_memory_limit * kRedLimitFactor) / shard_set->size();
```

Comment:
i think when eviction and tiering are both enabled, it gets tricky:

To me, eviction_redline shall be calculated correspondingly:

1) if tiering is not enabled, it is computed in the current way. 
2) otherwise, it should be calculated as:

```
ssize_t eviction_redline = (max_capacity_limit * kRedLimitFactor) / shard_set->size();
```

where `max_capacity_limit` should be the sum of DRAM and tiering file capacity (e.g. `tiered_max_file_size`)

In general, I think eviction shall not happen before offloading, and we need to add code to check this condition.

Reply:
I did not touch eviction logic for tiering, it should be handled but it is not in this PR.
I am not sure we will have the same eviction logic as we have today for tiering

Reply:
The practical implication is that eviction policy should delete items if available RAM is low like we do today but it should  delete both offloaded and hot items.

Reply:
it depends on how you want to define tiering: if storage is only an extension of memory, then it is considered as memory too, so the whole dram and storage form a single piece of cache, and eviction shall cover data in cache i.e. dram + storage. 

But if one day storage becomes really persistent, it will make sense to use what you suggested